src/attributes/utils/data.ts

```diff
@@ -61,6 +81,22 @@ function getSimpleAttributeData(
   };
 }
 
+function getSwatchAttributeData(
+  data: AttributePageFormData,
+  values: AttributeValueEditDialogFormData[]
+) {
+  return {
+    ...data,
+    metadata: undefined,
+    privateMetadata: undefined,
+    storefrontSearchPosition: parseInt(data.storefrontSearchPosition, 10),
+    values: values.map(({ fileUrl, name, value, contentType }) => ({
+      name,
+      ...(fileUrl ? { fileUrl, contentType } : { value })
```

Comment:
Same here, what do you think about making util for it e.g. `getAttributeValueTypeFields(attributeValue)`?